### ISSUE TITLE
error with expr.gen_torch_exp()

### ISSUE THREAD

**Issue by pauldebdeep9** [stellatogrp/lropt#42]
Here is my environment:

name: lropt
channels:
  - gurobi
  - conda-forge
  - defaults
dependencies:
  - anyio=4.2.0=py311hecd8cb5_0
  - appnope=0.1.2=py311hecd8cb5_1001
  - argon2-cffi=21.3.0=pyhd3eb1b0_0
  - argon2-cffi-bindings=21.2.0=py311h6c40b1e_0
  - asttokens=2.0.5=pyhd3eb1b0_0
  - async-lru=2.0.4=py311hecd8cb5_0
  - attrs=23.1.0=py311hecd8cb5_0
  - babel=2.11.0=py311hecd8cb5_0
  - beautifulsoup4=4.12.3=py311hecd8cb5_0
  - blas=1.0=openblas
  - bleach=4.1.0=pyhd3eb1b0_0
  - bottleneck=1.3.7=py311hb3a5e46_0
  - brotli=1.0.9=h6c40b1e_8
  - brotli-bin=1.0.9=h6c40b1e_8
  - brotli-python=1.0.9=py311hcec6c5f_8
  - bzip2=1.0.8=h6c40b1e_6
  - ca-certificates=2024.7.2=hecd8cb5_0
  - certifi=2024.7.4=py311hecd8cb5_0
  - cffi=1.16.0=py311h6c40b1e_1
  - charset-normalizer=3.3.2=pyhd3eb1b0_0
  - clarabel=0.7.1=py311h4b074ad_0
  - comm=0.2.1=py311hecd8cb5_0
  - contourpy=1.2.0=py311ha357a0b_0
  - cvxpy=1.5.3=py311h6eed73b_0
  - cvxpy-base=1.5.3=py311hfdcbad3_0
  - cycler=0.11.0=pyhd3eb1b0_0
  - debugpy=1.6.7=py311hcec6c5f_0
  - decorator=5.1.1=pyhd3eb1b0_0
  - defusedxml=0.7.1=pyhd3eb1b0_0
  - ecos=2.0.12=py311hb9e55a9_0
  - et_xmlfile=1.1.0=py311hecd8cb5_0
  - executing=0.8.3=pyhd3eb1b0_0
  - fonttools=4.51.0=py311h6c40b1e_0
  - freetype=2.12.1=hd8bbffd_0
  - gurobi=11.0.2=py311_0
  - idna=3.7=py311hecd8cb5_0
  - intel-openmp=2023.1.0=ha357a0b_43548
  - ipykernel=6.28.0=py311hecd8cb5_0
  - ipython=8.25.0=py311hecd8cb5_0
  - jedi=0.19.1=py311hecd8cb5_0
  - jinja2=3.1.4=py311hecd8cb5_0
  - joblib=1.4.2=py311hecd8cb5_0
  - jpeg=9e=h46256e1_3
  - json5=0.9.6=pyhd3eb1b0_0
  - jsonschema=4.19.2=py311hecd8cb5_0
  - jsonschema-specifications=2023.7.1=py311hecd8cb5_0
  - jupyter-lsp=2.2.0=py311hecd8cb5_0
  - jupyter_client=8.6.0=py311hecd8cb5_0
  - jupyter_core=5.7.2=py311hecd8cb5_0
  - jupyter_events=0.10.0=py311hecd8cb5_0
  - jupyter_server=2.14.1=py311hecd8cb5_0
  - jupyter_server_terminals=0.4.4=py311hecd8cb5_1
  - jupyterlab=4.0.11=py311hecd8cb5_0
  - jupyterlab_pygments=0.1.2=py_0
  - jupyterlab_server=2.25.1=py311hecd8cb5_0
  - kiwisolver=1.4.4=py311hcec6c5f_0
  - lcms2=2.12=hf1fd2bf_0
  - lerc=3.0=he9d5cce_0
  - libblas=3.9.0=22_osx64_openblas
  - libbrotlicommon=1.0.9=h6c40b1e_8
  - libbrotlidec=1.0.9=h6c40b1e_8
  - libbrotlienc=1.0.9=h6c40b1e_8
  - libcblas=3.9.0=22_osx64_openblas
  - libcxx=17.0.6=heb59cac_3
  - libdeflate=1.17=hb664fd8_1
  - libffi=3.4.4=hecd8cb5_1
  - libgfortran=5.0.0=13_2_0_h97931a8_3
  - libgfortran5=13.2.0=h2873a65_3
  - liblapack=3.9.0=22_osx64_openblas
  - libopenblas=0.3.27=openmp_h8869122_1
  - libosqp=0.6.3=hcec6c5f_0
  - libpng=1.6.39=h6c40b1e_0
  - libqdldl=0.1.7=hcec6c5f_0
  - libsodium=1.0.18=h1de35cc_0
  - libsqlite=3.46.0=h1b8f9f3_0
  - libtiff=4.5.1=hcec6c5f_0
  - libwebp-base=1.3.2=h6c40b1e_0
  - libzlib=1.2.13=h87427d6_6
  - llvm-openmp=18.1.8=h15ab845_0
  - lz4-c=1.9.4=hcec6c5f_1
  - markupsafe=2.1.3=py311h6c40b1e_0
  - matplotlib=3.9.2=py311hecd8cb5_0
  - matplotlib-base=3.9.2=py311hc25a08b_0
  - matplotlib-inline=0.1.6=py311hecd8cb5_0
  - mistune=2.0.4=py311hecd8cb5_0
  - mkl=2023.2.0=h54c2260_50500
  - mkl-service=2.4.0=py311h6c40b1e_1
  - mkl_fft=1.3.10=py311h8fccfe8_1
  - mkl_random=1.2.7=py311h02ad33b_0
  - nbclient=0.8.0=py311hecd8cb5_0
  - nbconvert=7.10.0=py311hecd8cb5_0
  - nbformat=5.9.2=py311hecd8cb5_0
  - ncurses=6.4=hcec6c5f_0
  - nest-asyncio=1.6.0=py311hecd8cb5_0
  - notebook=7.0.8=py311hecd8cb5_2
  - notebook-shim=0.2.3=py311hecd8cb5_0
  - numexpr=2.8.7=py311h91b6869_0
  - numpy=1.26.4=py311h91b6869_0
  - numpy-base=1.26.4=py311hb3ec012_0
  - openjpeg=2.5.2=hbf2204d_0
  - openpyxl=3.1.5=py311h46256e1_0
  - openssl=3.0.14=h46256e1_0
  - osqp=0.6.3=py311hdb55bb0_0
  - overrides=7.4.0=py311hecd8cb5_0
  - packaging=24.1=py311hecd8cb5_0
  - pandas=2.2.2=py311he327ffe_0
  - pandocfilters=1.5.0=pyhd3eb1b0_0
  - parso=0.8.3=pyhd3eb1b0_0
  - pexpect=4.8.0=pyhd3eb1b0_3
  - pillow=10.4.0=py311h46256e1_0
  - pip=24.2=py311hecd8cb5_0
  - platformdirs=3.10.0=py311hecd8cb5_0
  - prometheus_client=0.14.1=py311hecd8cb5_0
  - prompt-toolkit=3.0.43=py311hecd8cb5_0
  - prompt_toolkit=3.0.43=hd3eb1b0_0
  - psutil=5.9.0=py311h6c40b1e_0
  - ptyprocess=0.7.0=pyhd3eb1b0_2
  - pure_eval=0.2.2=pyhd3eb1b0_0
  - pybind11-abi=5=hd3eb1b0_0
  - pycparser=2.21=pyhd3eb1b0_0
  - pygments=2.15.1=py311hecd8cb5_1
  - pyparsing=3.1.2=py311hecd8cb5_0
  - pysocks=1.7.1=py311hecd8cb5_0
  - python=3.11.0=he7542f4_1_cpython
  - python-dateutil=2.9.0post0=py311hecd8cb5_2
  - python-fastjsonschema=2.16.2=py311hecd8cb5_0
  - python-json-logger=2.0.7=py311hecd8cb5_0
  - python-tzdata=2023.3=pyhd3eb1b0_0
  - python_abi=3.11=5_cp311
  - pytz=2024.1=py311hecd8cb5_0
  - pyyaml=6.0.1=py311h6c40b1e_0
  - pyzmq=25.1.2=py311hcec6c5f_0
  - qdldl-python=0.1.7=py311hdb55bb0_0
  - readline=8.2=hca72f7f_0
  - referencing=0.30.2=py311hecd8cb5_0
  - requests=2.32.3=py311hecd8cb5_0
  - rfc3339-validator=0.1.4=py311hecd8cb5_0
  - rfc3986-validator=0.1.1=py311hecd8cb5_0
  - rpds-py=0.10.6=py311hf2ad997_0
  - scikit-learn=1.5.1=py311he327ffe_0
  - scipy=1.14.1=py311hb3ed397_0
  - scs=3.2.3=py311hb3a5e46_0
  - send2trash=1.8.2=py311hecd8cb5_0
  - setuptools=72.1.0=py311hecd8cb5_0
  - six=1.16.0=pyhd3eb1b0_1
  - sniffio=1.3.0=py311hecd8cb5_0
  - soupsieve=2.5=py311hecd8cb5_0
  - sqlite=3.45.3=h6c40b1e_0
  - stack_data=0.2.0=pyhd3eb1b0_0
  - tbb=2021.8.0=ha357a0b_0
  - terminado=0.17.1=py311hecd8cb5_0
  - threadpoolctl=3.5.0=py311h85bffb1_0
  - tinycss2=1.2.1=py311hecd8cb5_0
  - tk=8.6.14=h4d00af3_0
  - tornado=6.4.1=py311h46256e1_0
  - traitlets=5.14.3=py311hecd8cb5_0
  - typing-extensions=4.11.0=py311hecd8cb5_0
  - typing_extensions=4.11.0=py311hecd8cb5_0
  - tzdata=2024a=h04d1e81_0
  - unicodedata2=15.1.0=py311h6c40b1e_0
  - urllib3=2.2.2=py311hecd8cb5_0
  - wcwidth=0.2.5=pyhd3eb1b0_0
  - webencodings=0.5.1=py311hecd8cb5_1
  - websocket-client=1.8.0=py311hecd8cb5_0
  - wheel=0.43.0=py311hecd8cb5_0
  - xz=5.4.6=h6c40b1e_1
  - yaml=0.2.5=haf1e3a3_0
  - zeromq=4.3.5=hcec6c5f_0
  - zlib=1.2.13=h87427d6_6
  - zstd=1.5.5=hc035e20_2
  - pip:
      - absl-py==2.1.0
      - astunparse==1.6.3
      - cvxpylayers==0.1.6
      - diffcp==1.0.23
      - filelock==3.15.4
      - flatbuffers==24.3.25
      - fsspec==2024.6.1
      - gast==0.6.0
      - google-pasta==0.2.0
      - grpcio==1.64.1
      - h5py==3.11.0
      - keras==3.4.1
      - libclang==18.1.1
      - lropt==0.0.1.dev0
      - markdown==3.6
      - markdown-it-py==3.0.0
      - mdurl==0.1.2
      - ml-dtypes==0.3.2
      - mpmath==1.3.0
      - namex==0.0.8
      - networkx==3.3
      - opt-einsum==3.3.0
      - optree==0.12.1
      - protobuf==4.25.3
      - pybind11==2.13.5
      - rich==13.7.1
      - rsome==1.3.0
      - sympy==1.13.2
      - tensorboard==2.16.2
      - tensorboard-data-server==0.7.2
      - tensorflow==2.16.2
      - tensorflow-io-gcs-filesystem==0.37.1
      - termcolor==2.4.0
      - torch==2.2.2
      - torchaudio==2.2.2
      - torchvision==0.17.2
      - werkzeug==3.0.3
      - wrapt==1.16.0
prefix: /Users/debdeeppaul/opt/anaconda3/envs/lropt

While using the api i am encountering the following error:
![Screenshot 2024-09-26 at 1 29 15 PM](https://github.com/user-attachments/assets/b7bf4680-32f7-45db-9591-56e7fa118a00)

Thanks a lot for developing this awesome library. It's incredibly well written with seamless workflow. I guess there was some discussion on robustifying the expr.gen_torch_exp() involving torch and cvxpy. Has there any incompatibility between different versions of torch and cvxpy.

**Comment by AmitSolomonPrinceton** [comment]
Hello @pauldebdeep9,
Thank you for your kind words. We have recently merged a PR that overhauls the way torch expressions are generated. Could you kindly pull that and see if the issue persists?

**Comment by pauldebdeep9** [comment]
Hi @AmitSolomonPrinceton, works fine. 
This is an awesome library. This area of optimization is evolving very quickly. The usage of torch is quite appropriate. Hope to .witness widespread adoption.
Best, Debdeep
